sr_robot_launch/scripts/teach_mode_node

```diff
@@ -112,10 +110,6 @@ class TeachModeManager(object):
                 rospy.loginfo("Changing %s Control mode to FORCE", req.robot)
                 self.ctrl_helpers[
                     req.robot].change_force_ctrl_type(change_type_msg)
-                rospy.loginfo("Changing %s controllers to effort", req.robot)
-                self.ctrl_helpers[req.robot].change_hand_ctrl("effort")
-                rospy.loginfo(
```

Comment:
Maybe we could leave this one in? The trajectory controller starting/stopping with effort/position/trajectory mode isn't necessarily obvious

Reply:
This is the part which was moved to C++ so we should change controllers twice.

Reply:
does the c++ code notify about the trajectory controller, or just FORCE/PWM mode?

Reply:
C++ code and driver only knows about FORCE/PWM. sr_robot_msgs::ControlType has only values PWM, FORCE and QUERY (for asking what is the current mode) so trajectory controllers seems to be handled completely outside.

Reply:
but do we still get the messages in console?

Reply:
Yes, we get messages in console on CPP side. We get message per loaded controller: https://github.com/shadow-robot/sr-ros-interface-ethercat/blob/F%23SRC-4962_switch_controllers/sr_robot_lib/src/sr_motor_robot_lib.cpp#L1271 and a message per switched controller pair: https://github.com/shadow-robot/sr-ros-interface-ethercat/blob/F%23SRC-4962_switch_controllers/sr_robot_lib/src/sr_motor_robot_lib.cpp#L1285

Reply:
For example:
```
[ INFO] [1623751654.417274597]: Changing control type to TORQUE
[ INFO] [1623751654.432377858]:   No forward  friction compensation map for: rh_FFJ0
[ INFO] [1623751654.432458941]:   No backward friction compensation map for: rh_FFJ0
[ INFO] [1623751654.440381554]: Loaded controller: sh_rh_ffj0_effort_controller
[ INFO] [1623751654.445026070]: Switched from controller sh_rh_ffj0_position_controller to sh_rh_ffj0_effort_controller
[ INFO] [1623751654.453454615]:   No forward  friction compensation map for: rh_FFJ3
[ INFO] [1623751654.453540368]:   No backward friction compensation map for: rh_FFJ3
[ INFO] [1623751654.461527532]: Loaded controller: sh_rh_ffj3_effort_controller
[ INFO] [1623751654.466160697]: Switched from controller sh_rh_ffj3_position_controller to sh_rh_ffj3_effort_controller
[ INFO] [1623751654.473649840]:   No forward  friction compensation map for: rh_FFJ4
[ INFO] [1623751654.473748927]:   No backward friction compensation map for: rh_FFJ4
[ INFO] [1623751654.483558385]: Loaded controller: sh_rh_ffj4_effort_controller
[ INFO] [1623751654.488785418]: Switched from controller sh_rh_ffj4_position_controller to sh_rh_ffj4_effort_controller
[ INFO] [1623751654.495581683]:   No forward  friction compensation map for: rh_MFJ0
[ INFO] [1623751654.495657604]:   No backward friction compensation map for: rh_MFJ0
```

Reply:
Switching back to position:
```
[ INFO] [1623751679.394434953]: Changing control type to PWM
[ WARN] [1623751679.396337857]: Reseting PID for joints rh_FFJ1 and rh_FFJ2
[ INFO] [1623751679.396534413]: Switched from controller sh_rh_ffj0_effort_controller to sh_rh_ffj0_position_controller
[ WARN] [1623751679.397369551]: Reseting PID for joint  rh_FFJ3
[ INFO] [1623751679.397469262]: Switched from controller sh_rh_ffj3_effort_controller to sh_rh_ffj3_position_controller
[ WARN] [1623751679.398313928]: Reseting PID for joint  rh_FFJ4
[ INFO] [1623751679.398381349]: Switched from controller sh_rh_ffj4_effort_controller to sh_rh_ffj4_position_controller
[ WARN] [1623751679.399362202]: Reseting PID for joints rh_MFJ1 and rh_MFJ2
[ INFO] [1623751679.399472295]: Switched from controller sh_rh_mfj0_effort_controller to sh_rh_mfj0_position_controller
[ WARN] [1623751679.400314015]: Reseting PID for joint  rh_MFJ3
[ INFO] [1623751679.400390548]: Switched from controller sh_rh_mfj3_effort_controller to sh_rh_mfj3_position_controller
```

Reply:
Example for switching to teach mode again, so that controllers are already loaded:
```
[ INFO] [1623750900.825555938]: Changing control type to TORQUE
[ INFO] [1623750900.829867226]: Switched from controller sh_rh_ffj0_position_controller to sh_rh_ffj0_effort_controller
[ INFO] [1623750900.833564657]: Switched from controller sh_rh_ffj3_position_controller to sh_rh_ffj3_effort_controller
[ INFO] [1623750900.838102916]: Switched from controller sh_rh_ffj4_position_controller to sh_rh_ffj4_effort_controller
[ INFO] [1623750900.842946117]: Switched from controller sh_rh_mfj0_position_controller to sh_rh_mfj0_effort_controller
[ INFO] [1623750900.846884853]: Switched from controller sh_rh_mfj3_position_controller to sh_rh_mfj3_effort_controller
[ INFO] [1623750900.850762512]: Switched from controller sh_rh_mfj4_position_controller to sh_rh_mfj4_effort_controller
[ INFO] [1623750900.854876464]: Switched from controller sh_rh_rfj0_position_controller to sh_rh_rfj0_effort_controller
```

Reply:
What about position to trajectory and back?

Reply:
Trajectory controllers are handled with separate calls to `change_trajectory_ctrl` instead of removed `change_hand_ctrl`. I didn't change anything related to trajectory so I expect I didn't broke them to. Is there a way we could test if trajectory control works as expected?

Reply:
it works as expected, I'm just not sure all the notifications would still be there. If this PR was merged as-is, switching into or out of FORCE or POSITION/TRAJECTORY would produce the correct notifications, but switching between POSITION and TRAJECTORY would result in no notifications to the user. We should remove the lines you have suggested we remove here (as their notifications have been moved to robot lib), but the trajectory controller on/off stuff is removed from here but not added to robot lib, so maybe we should leave the trajectory controller notifications here

Reply:
Sorry that it took so long to understand what do you mean. Restored trajectory controller log message in 65bc1dac0dcffd447f5761e4bbd3c66dc3d8a450.